plugins/woocommerce-blocks/packages/components/text-input/validated-text-input.tsx

```diff
@@ -117,29 +121,39 @@ const ValidatedTextInput = forwardRef<
 
 				if (
 					inputObject.checkValidity() &&
-					customValidationRef.current( inputObject )
+					customValidationRef.current( inputObject ) &&
+					errorsHidden
```

Comment:
We only clear errors if we're showing errors, this seems to now fixes everything.